packages/vscode-extension/src/handlers.ts

```diff
@@ -176,6 +176,14 @@ export async function createNewProjectHandler(args?: any[]): Promise<Result<null
   return await runCommand(Stage.create);
 }
 
+export async function migrateV1ProjectHandler(args?: any[]): Promise<Result<null, FxError>> {
+  ExtTelemetry.sendTelemetryEvent(
+    TelemetryEvent.MigrateV1ProjectStart,
+    getTriggerFromProperty(args)
+  );
+  return await runCommand(Stage.migrateV1);
```

Comment:
What if user trigger this command from a V2 project?

Reply:
An option may be - make this command a hidden one and can only be triggered by our detection.

Reply:
This command may be also used in a react app. 
We need to prohibit users from migrating v2 projects in both cli and vsc.
Would you like to throw an error message when the app is a v2 app (exist a `.fx` folder)?  It can be implemented in core.

Reply:
I think it is fine. Maybe need to confirm with Junjie.

Reply:
we may can add a "custom when clause context" for this command. this command only available when open v1 project.

Reply:
The command is hidden in vscode if it is not a v1 project.

The same v1 project validation will be added into core in another PR.